WIP PR for SRVLOGIC-310: Document "Job Service" section in the OSL docs (#81945)
openshift/openshift-docs: serverless-logic/serverless-logic-supporting-services/serverless-logic-job-service.adoc

The Job service schedules and executes tasks in a cloud environment. Independent services implement these tasks, which can be initiated through any of the supported interaction modes, including HTTP calls or Knative Events delivery.

In {ServerlessLogicProductName}, the Job service is responsible for controlling the execution of the time-triggered actions. Therefore all the time-based states that you can use in a workflow, are handled by the interaction between the workflow and the Job service.

For example, every time the workflow execution reaches a state with a configured timeout, a corresponding job is created in the Job service, and when the timeout is met, a http callback is executed to notify the workflow.

The main goal of the Job service is to manage active jobs, such as scheduled jobs that need to be executed. Once a job reaches its final state, the Job service removes it. To retain jobs information in a permanent repository, the Job service produce status change events, that can be recorded by an external service, such as the Data Index Service.

[NOTE]
====

> When you are using the {ServerlessOperatorName} to deploy workflows, there is no need to manually install or configure the Job service. The Operator handles these tasks automatically and manages all necessary configurations for each workflow to connect with it.

Reviewer: Consider rewording it for clarity. For example:
```suggestion
You do not need to manually install or configure the Job service if you are using the {ServerlessOperatorName} to deploy workflows. The Operator handles these tasks automatically and manages all necessary configurations for each workflow to connect with it.
```

The Job service schedules and executes tasks in a cloud environment. Independent services implement these tasks, which can be initiated through any of the supported interaction modes, including HTTP calls or Knative Events delivery.

In {ServerlessLogicProductName}, the Job service is responsible for controlling the execution of the time-triggered actions. Therefore all the time-based states that you can use in a workflow, are handled by the interaction between the workflow and the Job service.

> For example, every time the workflow execution reaches a state with a configured timeout, a corresponding job is created in the Job service, and when the timeout is met, a http callback is executed to notify the workflow.

Reviewer 2: Maybe we should add an example workflow here @wmedvede

Author: Thank you!

> [NOTE]
> ====
> If the project is not configured to use the Job Service, all time-based actions will use an in-memory implementation of that service. However, this setup must not be used in production, since every time the application is restarted, all the timers are lost, making it unsuitable for serverless architectures where applications might scale to zero at any time, etc.
> ====

Reviewer 3: ```suggestion
```

Author: removed the note

> The main goal of the Job service is to manage active jobs, such as scheduled jobs that need to be executed. Once a job reaches its final state, the Job service removes it. Ff you want to retain job information in a permanent repository, you can configure the Job service to produce status change events. This allows you to extend the jobs life span completion.

Author: Added!

[NOTE]
====

> When you are using the {ServerlessOperatorName} to deploy workflows, there is no need to manually install or configure the Job service. The Operator handles these tasks automatically and manages all necessary configurations for each workflow to connect with it.

Reviewer 3: @kaldesai I think that in this note, we should add a link to the upcoming supporting services section.
So, we have so we have something like this:

.... all necessary configurations for each workflow to connect with it, **see**"  :point_left: and here we add the link

====

> include::modules/serverless-logic-job-service-custom-execution.adoc[leveloffset=+1]

Reviewer 3: @kaldesai So, after discussion, we don't make available this jobs-service-custom-execution section. 
But, the comments applied in the particular doc serverless-logic-job-service-custom-execution.adoc are still good.
I think we can incorporate them and keep in a template for the future.

Author: Sure!

> include::modules/serverless-logic-job-service-common-config.adoc[leveloffset=+1]

Reviewer 3: @kaldesai Similar comment for the serverless-logic-job-service-common-config.adoc document. Comments are good, but we don't make it available by now.

> include::modules/serverless-logic-job-service-postgresql-ephemeral-config.adoc[leveloffset=+1]

Reviewer 3: @kaldesai Similar for the serverless-logic-job-service-postgresql-ephemeral-config.adoc

> include::modules/serverless-logic-job-service-eventing-api-config.adoc[leveloffset=+1]

Reviewer 3: @kaldesai Similar for the serverless-logic-job-service-eventing-api-config.adoc
